brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.brave;
+
+import static com.linecorp.armeria.common.brave.RequestContextCurrentTraceContext.ensureScopeUsesRequestContext;
+
+import java.util.function.Function;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.client.tracing.HttpTracingClient;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.brave.RequestContextCurrentTraceContext;
+
+import brave.Tracing;
+import brave.http.HttpTracing;
+
+/**
+ * Decorates a {@link Client} to trace outbound {@link HttpRequest}s using
+ * <a href="https://github.com/openzipkin/brave">Brave</a>.
+ *
+ * <p>This decorator puts trace data into HTTP headers. The specifications of header names and its values
+ * correspond to <a href="http://zipkin.io/">Zipkin</a>.
+ */
+public final class BraveClient extends HttpTracingClient {
```

Comment:
I think we might have talked a bit about this, but now even moreso, it seems like `HttpTracingClient` should extend `BraveClient`. Even for just a deprecation period, it seems weird for the `brave` artifact to have to depend on a deprecated `zipkin` artifact.

Reply:
Yes it looks very weird that the `brave` artifact depends on `zipkin` artifact. :-)
But, I did it to reduce duplicate code in `BraveClient` and `BraveService`.
A user used to use three public classes which are `HttpTracingClient`, `HttpTracingService` and `RequestContextCurrentTraceContext`.
So I added `BraveClient`, `BraveService` and `RequestContextCurrentTraceContext` (different package) which correspond to them.

If I want to make `BraveClient` to extend `HttpTracingClient` or vice versa, I had to add this creepy constructor:
```
protected HttpTracingService(Service<HttpRequest, HttpResponse> delegate, Tracing tracing,
                             BiConsumer<RequestContext, RequestContext> traceContextPropagator) {...}
```
because `RequestContextCurrentTraceContext.copy()` method is used in the `execute()`.
I didn't want to add this constuctor to `brave` module, so I chose to add it to `zipkin`. That's the reason.

Perhaps I just copy the code from `HttpTracingClient` and `HttpTracingService` to the new module like I did in the first place.
What do you think?

Reply:
Ah thanks for the details, I understand the issue now. I might have a slight preference to copying, and we just delete the zipkin folder once deprecation is done, but am ok with this too.

Reply:
I prefer copying too. :-) Let me do that. 😄 Thanks!